src/mapml/handlers/QueryHandler.js

```diff
@@ -104,74 +104,72 @@ export var QueryHandler = L.Handler.extend({
-                  imagePath: M.detectImagePath(map.getContainer())
-              });
-              f.addTo(map);
+      function handleMapMLResponse(mapml, loc) {
+        var parser = new DOMParser(),
+            mapmldoc = parser.parseFromString(mapml, "application/xml"),
+            f = M.mapMlFeatures(mapmldoc, {
+            // pass the vector layer a renderer of its own, otherwise leaflet
+            // puts everything into the overlayPane
+            renderer: L.svg(),
+            // pass the vector layer the container for the parent into which
+            // it will append its own container for rendering into
+            pane: container,
+            //color: 'yellow',
+            // instead of unprojecting and then projecting and scaling,
+            // a much smarter approach would be to scale at the current
+            // zoom
+            //coordsToLatLng: _coordsToLatLng,
+            _leafletLayer: layer,
+            imagePath: M.detectImagePath(map.getContainer()),
+            query: true,
+        });
+        f.addTo(map);
 
-              var c = document.createElement('iframe');
-              c.csp = "script-src 'none'";
-              c.style = "border: none";
-              c.srcdoc = mapmldoc.querySelector('feature properties').innerHTML;
+        let div = L.DomUtil.create("div", "mapml-popup-content"),
+            c = L.DomUtil.create("iframe");
+        c.csp = "script-src 'none'";
+        c.style = "border: none";
+        c.srcdoc = `<meta http-equiv="content-security-policy" content="script-src 'none';">` + mapmldoc.querySelector('feature properties').innerHTML;
+        div.appendChild(c);
+        // passing a latlng to the popup is necessary for when there is no
+        // geometry / null geometry
+        layer._totalFeatureCount = mapmldoc.querySelectorAll("feature").length;
+        layer.bindPopup(div, popupOptions).openPopup(loc);
+        layer.on('popupclose', function() {
+            map.removeLayer(f);
+        });
+        f.showPaginationFeature({i: 0, popup: layer._popup});
 
-              // passing a latlng to the popup is necessary for when there is no
-              // geometry / null geometry
-              layer.bindPopup(c, popupOptions).openPopup(loc);
-              layer.on('popupclose', function() {
-                  map.removeLayer(f);
-              });
-          });
       }
-      function handleOtherResponse(response, layer, loc) {
-          return response.text().then(text => {
-              var c = document.createElement('iframe');
-              c.csp = "script-src 'none'";
-              c.style = "border: none";
-              c.srcdoc = text;
-              layer.bindPopup(c, popupOptions).openPopup(loc);
-          });
+      function handleOtherResponse(text, layer, loc) {
+        let div = L.DomUtil.create("div", "mapml-popup-content"),
+            c = L.DomUtil.create("iframe");
+        c.csp = "script-src 'none'";
```

Comment:
I don't think there's a need to set the `csp` attribute anymore, I suggest removing this line.